source/common/common/callback_impl.h

```diff
@@ -61,8 +63,8 @@ template <typename... CallbackArgs> class CallbackManager {
                         [handle](const CallbackHolder& holder) -> bool {
                           return handle == &holder;
                         }) != callbacks_.end());
-    callbacks_.remove_if(
-        [handle](const CallbackHolder& holder) -> bool { return handle == &holder; });
+    auto it = static_cast<CallbackHolder*>(handle)->it_;
+    callbacks_.erase(it);
```

Comment:
can we convert this to a flat_hash_map so we dont need to store the iterator and possibly have a better readability?

Reply:
I thought about using a hash_map, but a hash_map increase the time complexity of almost each operation when compared to list. The performance gain from O(1) removal does not necessarily compensate for the time spent in hash_map operation such as insertion and iteration, which defeat our initial purpose. Can we add some comments to make the code more readable?

Reply:
cost by hashmap is not a big deal. This callback manager is not "heavily" used anyway.
I don't have strong opinion here, but readability usually weigh in more than minor performance improvement.